### PR TITLE
[java] Selenium Manager generates output from argument list 

### PR DESCRIPTION
**Pull request by titusfortner**
### Description
* Have `DriverFinder` convert the `Capabilities` into the arguments required by Selenium Manager binary
* Deprecate existing `getPath()` method

### Motivation and Context
* Addresses Java portion of #13022 
* The method name `getPath()` is no longer an accurate one

### Updates
Updates:

* Changes `getPath()` to `getResult()` in `DriverFinder` as well. The name isn't great, but it is the name of the current class and I can't think of anything better. (`getPaths()`? `SeleniumManagerOutput()`?) 
* `browserVersion` is set to `null` in the 3 applicable driver classes instead of DriverFinder, and only if there is a browser path.
* Removing instead of deprecating the methods that are no longer used
* File validations are done in the Results class
* No longer a specific "Selenium Manager" error, the specific errors in SM class will bubble up anyway, and it makes the try/catch easier.

### Status
* Do we want a different name than `getResult()`?
* I'm going to add tests still

### CODE REVIEW

**Comment by titusfortner** [comment]
Do I need to deprecate, or can I just remove because the class is annotated as `@beta`?

**Comment by titusfortner** [comment]
and should we also rename `DriverFinder`'s `getPath()` method? I think `DriverFinder` still makes sense as a class name, since the browser information is related to driver finding, but the return value is no longer a path.

**Review by diemol — APPROVED** [review]
I would delete since it is market as beta.

My only concern is that we do not have enough tests to properly verify this refactor. Haven't we had the situation in the past where moving some arguments around ended up in some bugs?

**Comment by titusfortner** [comment]
Tests are a good idea. 😂

File: java/src/org/openqa/selenium/remote/service/DriverFinder.java
Comment (bonigarcia) [inline java/src/org/openqa/selenium/remote/service/DriverFinder.java]: Are these capabilities always mutable? Maybe it is, I'm asking just to ensure.

Comment (titusfortner) [reply java/src/org/openqa/selenium/remote/service/DriverFinder.java]: Yes, all the browser option classes inherit from mutable capabilities class.

Comment (titusfortner) [reply java/src/org/openqa/selenium/remote/service/DriverFinder.java]: That said, looking at other code, it looks like we create a new presistent capabilities object rather than mutating. So I'm thinking I'll refactor this.

Comment (titusfortner) [reply java/src/org/openqa/selenium/remote/service/DriverFinder.java]: Yeah, if this is parsed where DriverFinder is called, it doesn't need and cast and it can be limited it to the places it actually applies.

File: java/src/org/openqa/selenium/manager/SeleniumManagerOutput.java
Comment (diemol) [inline java/src/org/openqa/selenium/manager/SeleniumManagerOutput.java]: Why do we need this logic here? If we need it, shouldn't it be inside Selenium Manager?

**Review by diemol — CHANGES_REQUESTED** [review]

File: java/src/org/openqa/selenium/manager/SeleniumManagerOutput.java
Comment (titusfortner) [reply java/src/org/openqa/selenium/manager/SeleniumManagerOutput.java]: I'm pulling in checks for what user is passing in as well so they get the same logging and error messaging.

**Comment by titusfortner** [comment]
Ok, refactored along the lines of Simon's suggestions, added tests, they pass, I'm generally happy with the naming I think right now, and it's matching Ruby & Python PRs. This can be merged once approved.

File: java/src/org/openqa/selenium/chrome/ChromeDriver.java
Comment (diemol) [inline java/src/org/openqa/selenium/chrome/ChromeDriver.java:102]: But why do we want to remove the `browserVersion`?

Comment (titusfortner) [reply java/src/org/openqa/selenium/chrome/ChromeDriver.java:102]: It's necessary in Firefox because Firefox does the compliant thing and fails if it doesn't match (e.g., "stable" will fail). Chrome it isn't technically necessary, but if Selenium Manager has matched the version and found the exact browser to use, sending the driver the version number is unnecessary.

Comment (titusfortner) [reply java/src/org/openqa/selenium/chrome/ChromeDriver.java:102]: Like, per the spec, Chrome *should fail if the browser versions do not match, but it does not, but I figured the code should handle it in case it became spec compliant.

Comment (diemol) [reply java/src/org/openqa/selenium/chrome/ChromeDriver.java:102]: Understood. We already do this in the Grid, and it was missing here. Thanks.

**Review by diemol — APPROVED** [review]
👍

File: java/src/org/openqa/selenium/chrome/ChromeDriver.java
Comment (titusfortner) [reply java/src/org/openqa/selenium/chrome/ChromeDriver.java:102]: It was being done in Driver Finder, this moves it up a level to the place that actually needs it.

**Review by diemol — APPROVED** [review]

**Pull request merged**
